src/main/java/io/fabric8/jenkins/openshiftsync/BuildSyncRunListener.java

```diff
@@ -250,38 +251,31 @@ private void upsertBuild(Run run, RunExt wfRunExt) {
     String logsConsoleUrl = joinPaths(buildUrl, "/console");
     String logsBlueOceanUrl = null;
     try {
-        // while we support Jenkins v1 (it is being deprecated in openshift 3.6), need to get at
-        // blueocean plugins via reflection;
-        // On those plugins specifically, there are utility functions in the blueocean-dashboard plugin which construct 
-        // this entire URI; however, attempting to pull that in as a maven dependency was untenable from an injected test perspective;
-        // the blueocean-rest-impl plugin was possible though, and the organization piece was in fact the only one
-        // that was missing for use to construct the entire URL manually.
-        // But with reflection, we can leverage the blueocean-dashboard logic.  Doing so here, but have left the 
-        // blueocean-rest-impl plugin usage as a comment for future reference if we move off of reflection.
+        // there are utility functions in the blueocean-dashboard plugin which construct 
+        // the entire blueocean URI; however, attempting to pull that in as a maven dependency was untenable from an injected test perspective;
+        // so we are leveraging reflection;
         Jenkins jenkins = Jenkins.getInstance();
         // NOTE, the excessive null checking is to keep `mvn findbugs:gui` quiet
         if (jenkins != null) {
             PluginManager pluginMgr = jenkins.getPluginManager();
             if (pluginMgr != null) {
                 ClassLoader cl = pluginMgr.uberClassLoader;
                 if (cl != null) {
-                    Class weburlbldr = cl.loadClass("io.jenkins.blueocean.BlueOceanWebURLBuilder");
-                    Method toBlueOceanURLMethod = weburlbldr.getMethod("toBlueOceanURL", hudson.model.ModelObject.class);
-                    Object blueOceanURI = toBlueOceanURLMethod.invoke(null, run);
-                    logsBlueOceanUrl = joinPaths(rootUrl, blueOceanURI.toString());
+                    Class weburlbldr = cl.loadClass("org.jenkinsci.plugins.blueoceandisplayurl.BlueOceanDisplayURLImpl");
+                    Constructor ctor = weburlbldr.getConstructor();
+                    Object displayURL = ctor.newInstance();
+                    Method getRunURLMethod = weburlbldr.getMethod("getRunURL", hudson.model.Run.class);
+                    Object blueOceanURI = getRunURLMethod.invoke(displayURL, run);
+                    logsBlueOceanUrl = blueOceanURI.toString();
+                    logsBlueOceanUrl = logsBlueOceanUrl.replaceAll("http://unconfigured-jenkins-location/", "");
+                    if (logsBlueOceanUrl.startsWith("http://") || logsBlueOceanUrl.startsWith("https://"))
+                        // still normalize string
+                        joinPaths("", logsBlueOceanUrl);
```

Comment:
dah ... last minute change that wasn't as simple as I thought :-) ... will fix in a moment